CoffeeShop.js

```diff
@@ -0,0 +1,106 @@
+class File {
+	static Exists(filePath) {
+		if (require("fs").existsSync(filePath))
+			return true;
+
+		return false;
+	}
+
+	static ReadAllLines(filePath) {
+		return require("fs").readFileSync(filePath).toString().split('\n')
+	}
+}
+
+function ParseCSV(filePath) {
+	var rawData = File.ReadAllLines(filePath);
+	var arrData = Array.from(Array(rawData.length), () => new Array(3));
+
+	try {
+		for (var i = 0; i < rawData.length; i++) {
+			if ((rawData[i].split(',').length - 1) != 2)
+				throw new Error(`Corrupted CSV template (check for the commas) -> row [${i + 1}]`);
+
+			// Retrieves the coffee shop name
+			arrData[i][0] = rawData[i].slice(0, rawData[i].indexOf(','));
+
+			// Retrieves the X user coordinate
+			arrData[i][1] = rawData[i].slice(rawData[i].indexOf(',') + 1, rawData[i].lastIndexOf(','));
+
+			if (isNaN(arrData[i][1]))
+				throw new Error(`Invalid [X] coordinate -> row [${i + 1}]`);
+			// Retrieves the Y user coordinate
+			arrData[i][2] = rawData[i].slice(rawData[i].lastIndexOf(',') + 1);
+
+			if (isNaN(arrData[i][2]))
+				throw new Error(`Invalid [Y] coordinate -> row [${i + 1}]`);
+		}
+	} catch(e) {
+		throw new Error(`Can't parse the selected CSV file!\n   > : {${e}}`);
+	}
+
+	return arrData;
+}
+
+function CalculateDistance(x, y, x1, y1) {
+  const p = 0.017453292519943295; // Math.PI / 180
+  const c = Math.cos;
+  const a = 0.5 - c((x1 - x) * p) / 2 + 
+          c(x * p) * c(x1 * p) * 
+          (1 - c((y1 - y) * p)) / 2;
+
+  return 12742 * Math.asin(Math.sqrt(a)); // 2 * R; R = 6371 km
```

Comment:
From the readme:
```
Assume all coordinates lie on a plane.
```

Reply:
I've fixed the formula which is used to retrieve the distance between two points on a flat surface